debezium-core/src/main/java/io/debezium/transforms/tracing/ActivateOpenTracingSpan.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.tracing;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.SmtManager;
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.util.GlobalTracer;
+
+/**
+ * This SMT enables integration with an OpenTracing system.
+ * The SMT creates a OpenTracing span and enriches it with metadata from envelope and source info block.<br/>
+ * It is possible to connect the span to a parent span created by a business application.
+ * The application then needs to export its OpenTracing active span into a database field.
+ * The SMT looks for a predefined field name in the {@code after} block
+ * and when found it extracts the parent span from it.<br>
+ * 
+ * See {@link EventDispatcher} for example of such implementation.
+ * 
+ *
+ * @param <R> the subtype of {@link ConnectRecord} on which this transformation will operate
+ * @author Jiri Pechanec
+ */
+public class ActivateOpenTracingSpan<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivateOpenTracingSpan.class);
+
+    private static final String DEFAULT_OPENTRACING_SPAN_VALUE = "opentracingspan";
+    private static final String DEFAULT_OPENTRACING_OPERATION_NAME = "debezium";
+
+    private static final String TAG_PREFIX = "debezium.";
+
+    private static final Field OPENTRACING_SPAN_FIELD = Field.create("opentracing.span.field")
+            .withDisplayName("OpenTracing Span Field")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDefault(DEFAULT_OPENTRACING_SPAN_VALUE)
+            .withDescription("The name of the field notaining java.util.Properties representation of OpenTracing span.");
```

Comment:
```suggestion
            .withDescription("The name of the field containing java.util.Properties representation of OpenTracing span.");
```
"notaining" should be a word though, meaning "not containing" ;)

Reply:
I'd also avoid mentioning the Java type; rather "map" in OT terms?

Reply:
@jpkrohling Yes, exactly.
@gunnarmorling That's intentional. We are using `java.util.Properties` to read and write the data so I want to impose this as an API for applications that would like to support tracing with CDC and don't use our event router. This brings me to the question I forgot to raise one issue. We have `io.debezium.transforms.tracing.DebeziumTextMap` present in both of outbox and core. How about introducing it into `debezium-api`. It will be shared by both modules and any 3rd party app that would want to use opentracing could use this API class. Than we could turn the `java.util.Properties` into an impletmentation detail and hide it from the user. It would require opentracing dependency for debezium-api but we coudl make it safely `provided` so there want be any risk of dependency bloating.